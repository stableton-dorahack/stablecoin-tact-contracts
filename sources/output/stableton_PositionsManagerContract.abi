{"name":"PositionsManagerContract","types":[{"name":"StateInit","header":null,"fields":[{"name":"code","type":{"kind":"simple","type":"cell","optional":false}},{"name":"data","type":{"kind":"simple","type":"cell","optional":false}}]},{"name":"Context","header":null,"fields":[{"name":"bounced","type":{"kind":"simple","type":"bool","optional":false}},{"name":"sender","type":{"kind":"simple","type":"address","optional":false}},{"name":"value","type":{"kind":"simple","type":"int","optional":false,"format":257}},{"name":"raw","type":{"kind":"simple","type":"slice","optional":false}}]},{"name":"SendParameters","header":null,"fields":[{"name":"bounce","type":{"kind":"simple","type":"bool","optional":false}},{"name":"to","type":{"kind":"simple","type":"address","optional":false}},{"name":"value","type":{"kind":"simple","type":"int","optional":false,"format":257}},{"name":"mode","type":{"kind":"simple","type":"int","optional":false,"format":257}},{"name":"body","type":{"kind":"simple","type":"cell","optional":true}},{"name":"code","type":{"kind":"simple","type":"cell","optional":true}},{"name":"data","type":{"kind":"simple","type":"cell","optional":true}}]},{"name":"Deploy","header":2490013878,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"DeployOk","header":2952335191,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"ChangeOwner","header":256331011,"fields":[{"name":"newOwner","type":{"kind":"simple","type":"address","optional":false}}]},{"name":"DepositCollateralMessage","header":299872568,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}}]},{"name":"WithdrawCollateralMessage","header":1747572062,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"debtRate","type":{"kind":"simple","type":"int","optional":false,"format":257}}]},{"name":"WithdrawStablecoinMessage","header":4082453719,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"debtRate","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"RepayStablecoinMessage","header":1854010438,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"debtRate","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"DoWithdrawCollateralMessage","header":3285005247,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"IncreaseTotalStableMessage","header":2517198105,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"DecreaseTotalStableMessage","header":3984666698,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"fees","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"MintFeesMessage","header":2638369821,"fields":[{"name":"to","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}}]},{"name":"JettonData","header":null,"fields":[{"name":"totalSupply","type":{"kind":"simple","type":"int","optional":false,"format":257}},{"name":"mintable","type":{"kind":"simple","type":"bool","optional":false}},{"name":"owner","type":{"kind":"simple","type":"address","optional":false}},{"name":"content","type":{"kind":"simple","type":"cell","optional":true}},{"name":"walletCode","type":{"kind":"simple","type":"cell","optional":false}}]},{"name":"JettonWalletData","header":null,"fields":[{"name":"balance","type":{"kind":"simple","type":"int","optional":false,"format":257}},{"name":"owner","type":{"kind":"simple","type":"address","optional":false}},{"name":"master","type":{"kind":"simple","type":"address","optional":false}},{"name":"walletCode","type":{"kind":"simple","type":"cell","optional":false}}]},{"name":"SetUserPositionDependecyMessage","header":4030201083,"fields":[{"name":"stablecoinMasterAddress","type":{"kind":"simple","type":"address","optional":false}},{"name":"positionsManagerAddress","type":{"kind":"simple","type":"address","optional":false}}]},{"name":"SetPositionIdMessage","header":1245051344,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"positionId","type":{"kind":"simple","type":"int","optional":false,"format":257}}]},{"name":"NewPositionIdMessage","header":225756565,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}}]},{"name":"SetPositionAddressMessage","header":1571217046,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"position","type":{"kind":"simple","type":"address","optional":false}}]},{"name":"MintMessage","header":2419415474,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}}]},{"name":"RepayBurnMessage","header":2959459244,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"fees","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}}]},{"name":"RepayBurnNotification","header":2355055729,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"fees","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}}]},{"name":"StablecoinBurnedMessage","header":3929120156,"fields":[{"name":"user","type":{"kind":"simple","type":"address","optional":false}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"fees","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}}]},{"name":"TokenTransfer","header":260734629,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"destination","type":{"kind":"simple","type":"address","optional":false}},{"name":"responseDestination","type":{"kind":"simple","type":"address","optional":true}},{"name":"customPayload","type":{"kind":"simple","type":"cell","optional":true}},{"name":"forwardTonAmount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"forwardPayload","type":{"kind":"simple","type":"slice","optional":false,"format":"remainder"}}]},{"name":"TokenTransferInternal","header":395134233,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"from","type":{"kind":"simple","type":"address","optional":false}},{"name":"responseAddress","type":{"kind":"simple","type":"address","optional":true}},{"name":"forwardTonAmount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"forwardPayload","type":{"kind":"simple","type":"slice","optional":false,"format":"remainder"}}]},{"name":"TokenNotification","header":1935855772,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"from","type":{"kind":"simple","type":"address","optional":false}},{"name":"forwardPayload","type":{"kind":"simple","type":"slice","optional":false,"format":"remainder"}}]},{"name":"TokenBurn","header":1499400124,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"owner","type":{"kind":"simple","type":"address","optional":false}},{"name":"responseAddress","type":{"kind":"simple","type":"address","optional":true}}]},{"name":"TokenBurnNotification","header":2078119902,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}},{"name":"amount","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"owner","type":{"kind":"simple","type":"address","optional":false}},{"name":"responseAddress","type":{"kind":"simple","type":"address","optional":true}}]},{"name":"TokenExcesses","header":3576854235,"fields":[{"name":"queryId","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]},{"name":"TokenUpdateContent","header":201882270,"fields":[{"name":"content","type":{"kind":"simple","type":"cell","optional":true}}]},{"name":"PositionState","header":null,"fields":[{"name":"collateral","type":{"kind":"simple","type":"uint","optional":false,"format":"coins"}},{"name":"debt","type":{"kind":"simple","type":"uint","optional":false,"format":64}}]}],"receivers":[{"receiver":"internal","message":{"kind":"typed","type":"DepositCollateralMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"NewPositionIdMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"WithdrawStablecoinMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"IncreaseTotalStableMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"RepayStablecoinMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"DecreaseTotalStableMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"WithdrawCollateralMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"DoWithdrawCollateralMessage"}},{"receiver":"internal","message":{"kind":"typed","type":"Deploy"}}],"getters":[{"name":"lastPositionId","arguments":[],"returnType":{"kind":"simple","type":"int","optional":false,"format":257}},{"name":"balance","arguments":[],"returnType":{"kind":"simple","type":"int","optional":false,"format":257}}],"errors":{"2":{"message":"Stack undeflow"},"3":{"message":"Stack overflow"},"4":{"message":"Integer overflow"},"5":{"message":"Integer out of expected range"},"6":{"message":"Invalid opcode"},"7":{"message":"Type check error"},"8":{"message":"Cell overflow"},"9":{"message":"Cell underflow"},"10":{"message":"Dictionary error"},"13":{"message":"Out of gas error"},"32":{"message":"Method ID not found"},"34":{"message":"Action is invalid or not supported"},"37":{"message":"Not enough TON"},"38":{"message":"Not enough extra-currencies"},"128":{"message":"Null reference exception"},"129":{"message":"Invalid serialization prefix"},"130":{"message":"Invalid incoming message"},"131":{"message":"Constraints error"},"132":{"message":"Access denied"},"133":{"message":"Contract stopped"},"134":{"message":"Invalid argument"},"135":{"message":"Code of a contract was not found"},"136":{"message":"Invalid address"},"5576":{"message":"not from gateKeeper"},"15032":{"message":"not from stablecoin master"},"16059":{"message":"Invalid value"},"22230":{"message":"Already set"},"31797":{"message":"debt less than repay amount"},"53160":{"message":"not from positions manager"},"61910":{"message":"not from positionsManager"},"63577":{"message":"withdrawal amount more than position has"}}}