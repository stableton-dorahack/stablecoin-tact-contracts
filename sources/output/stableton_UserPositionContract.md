# TACT Compilation Report
Contract: UserPositionContract
BOC Size: 1678 bytes

# Types
Total Types: 25

## StateInit
TLB: `_ code:^cell data:^cell = StateInit`
Signature: `StateInit{code:^cell,data:^cell}`

## Context
TLB: `_ bounced:bool sender:address value:int257 raw:^slice = Context`
Signature: `Context{bounced:bool,sender:address,value:int257,raw:^slice}`

## SendParameters
TLB: `_ bounce:bool to:address value:int257 mode:int257 body:Maybe ^cell code:Maybe ^cell data:Maybe ^cell = SendParameters`
Signature: `SendParameters{bounce:bool,to:address,value:int257,mode:int257,body:Maybe ^cell,code:Maybe ^cell,data:Maybe ^cell}`

## Deploy
TLB: `deploy#946a98b6 queryId:uint64 = Deploy`
Signature: `Deploy{queryId:uint64}`

## DeployOk
TLB: `deploy_ok#aff90f57 queryId:uint64 = DeployOk`
Signature: `DeployOk{queryId:uint64}`

## ChangeOwner
TLB: `change_owner#0f474d03 newOwner:address = ChangeOwner`
Signature: `ChangeOwner{newOwner:address}`

## DepositCollateralMessage
TLB: `deposit_collateral_message#11dfb138 user:address amount:coins = DepositCollateralMessage`
Signature: `DepositCollateralMessage{user:address,amount:coins}`

## WithdrawCollateralMessage
TLB: `withdraw_collateral_message#6829d55e user:address amount:coins debtRate:int257 = WithdrawCollateralMessage`
Signature: `WithdrawCollateralMessage{user:address,amount:coins,debtRate:int257}`

## WithdrawStablecoinMessage
TLB: `withdraw_stablecoin_message#f3554cd7 user:address amount:uint64 debtRate:uint64 = WithdrawStablecoinMessage`
Signature: `WithdrawStablecoinMessage{user:address,amount:uint64,debtRate:uint64}`

## RepayStablecoinMessage
TLB: `repay_stablecoin_message#6e81f446 user:address amount:uint64 debtRate:uint64 = RepayStablecoinMessage`
Signature: `RepayStablecoinMessage{user:address,amount:uint64,debtRate:uint64}`

## DoWithdrawCollateralMessage
TLB: `do_withdraw_collateral_message#c3cd33bf user:address amount:uint64 = DoWithdrawCollateralMessage`
Signature: `DoWithdrawCollateralMessage{user:address,amount:uint64}`

## IncreaseTotalStableMessage
TLB: `increase_total_stable_message#96096519 user:address amount:uint64 = IncreaseTotalStableMessage`
Signature: `IncreaseTotalStableMessage{user:address,amount:uint64}`

## DecreaseTotalStableMessage
TLB: `decrease_total_stable_message#ed81304a user:address amount:uint64 fees:uint64 = DecreaseTotalStableMessage`
Signature: `DecreaseTotalStableMessage{user:address,amount:uint64,fees:uint64}`

## MintFeesMessage
TLB: `mint_fees_message#9d42541d to:address amount:coins = MintFeesMessage`
Signature: `MintFeesMessage{to:address,amount:coins}`

## JettonData
TLB: `_ totalSupply:int257 mintable:bool owner:address content:Maybe ^cell walletCode:^cell = JettonData`
Signature: `JettonData{totalSupply:int257,mintable:bool,owner:address,content:Maybe ^cell,walletCode:^cell}`

## JettonWalletData
TLB: `_ balance:int257 owner:address master:address walletCode:^cell = JettonWalletData`
Signature: `JettonWalletData{balance:int257,owner:address,master:address,walletCode:^cell}`

## SetUserPositionDependecyMessage
TLB: `set_user_position_dependecy_message#f037fcfb stablecoinMasterAddress:address positionsManagerAddress:address = SetUserPositionDependecyMessage`
Signature: `SetUserPositionDependecyMessage{stablecoinMasterAddress:address,positionsManagerAddress:address}`

## SetPositionIdMessage
TLB: `set_position_id_message#4a35f9d0 user:address positionId:int257 = SetPositionIdMessage`
Signature: `SetPositionIdMessage{user:address,positionId:int257}`

## NewPositionIdMessage
TLB: `new_position_id_message#0d74c595 user:address = NewPositionIdMessage`
Signature: `NewPositionIdMessage{user:address}`

## SetPositionAddressMessage
TLB: `set_position_address_message#5da6de96 user:address position:address = SetPositionAddressMessage`
Signature: `SetPositionAddressMessage{user:address,position:address}`

## MintMessage
TLB: `mint_message#903559b2 user:address amount:coins = MintMessage`
Signature: `MintMessage{user:address,amount:coins}`

## BurnMessage
TLB: `burn_message#b625719b user:address amount:coins fees:coins = BurnMessage`
Signature: `BurnMessage{user:address,amount:coins,fees:coins}`

## RepayBurnNotification
TLB: `repay_burn_notification#563c6d94 user:address amount:coins = RepayBurnNotification`
Signature: `RepayBurnNotification{user:address,amount:coins}`

## StablecoinBurnedMessage
TLB: `stablecoin_burned_message#d2be4c6c user:address amount:uint64 fees:uint64 = StablecoinBurnedMessage`
Signature: `StablecoinBurnedMessage{user:address,amount:uint64,fees:uint64}`

## PositionState
TLB: `_ collateral:coins debt:uint64 = PositionState`
Signature: `PositionState{collateral:coins,debt:uint64}`

# Get Methods
Total Get Methods: 4

## getPositionUser

## getPositionId

## getPositionState

## owner

# Error Codes
2: Stack undeflow
3: Stack overflow
4: Integer overflow
5: Integer out of expected range
6: Invalid opcode
7: Type check error
8: Cell overflow
9: Cell underflow
10: Dictionary error
13: Out of gas error
32: Method ID not found
34: Action is invalid or not supported
37: Not enough TON
38: Not enough extra-currencies
128: Null reference exception
129: Invalid serialization prefix
130: Invalid incoming message
131: Constraints error
132: Access denied
133: Contract stopped
134: Invalid argument
135: Code of a contract was not found
136: Invalid address
15032: not from stablecoin master
22230: Already set
31797: debt less than repay amount
53160: not from positions manager
61910: not from positionsManager
63577: withdrawal amount more than position has