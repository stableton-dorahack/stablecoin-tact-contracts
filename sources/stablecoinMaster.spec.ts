import { Address, OpenedContract, Sender, toNano } from "ton";
import { ContractSystem, Logger, Tracker, Treasure } from "ton-emulator";
import { PositionsManagerContract } from "./output/stableton_PositionsManagerContract";
import { StablecoinMasterContract } from "./output/stableton_StablecoinMasterContract";
import { UserPositionContract } from "./output/stableton_UserPositionContract";
import { StablecoinJettonContract } from "./output/stableton_StablecoinJettonContract";

describe("StablecoinMasterContract", () => {
    let system: ContractSystem;
    let owner: Treasure;
    let user: Treasure;
    let userPosition: Treasure;
    let positionsManager: Treasure;
    let gateKeeper: Treasure;
    let jetton: Treasure;
    let stablecoinMasterContract: OpenedContract<StablecoinMasterContract>;
    let track: Tracker;
    let logger: Logger;

    beforeAll(async () => {
        system = await ContractSystem.create();
        owner = system.treasure("owner");
        user = system.treasure("user");
        gateKeeper = system.treasure("gateKeeper");
        positionsManager = system.treasure("positionsManager");
        userPosition = system.treasure("userPosition");
        jetton = system.treasure("jetton");

        stablecoinMasterContract = system.open(await StablecoinMasterContract.fromInit());

        track = system.track(stablecoinMasterContract.address);
        logger = system.log(stablecoinMasterContract.address);
    });

    it("should deploy correctly and set init state", async () => {
        await stablecoinMasterContract.send(owner, { value: toNano(1) }, { $$type: "Deploy", queryId: 0n });
        console.log("stablecoinMasterContract", stablecoinMasterContract.address);

        await system.run();
        expect(track.collect()).toMatchSnapshot();

        const masterData = await stablecoinMasterContract.getGetJettonData();
        expect(masterData.owner.toString()).toEqual(owner.address.toString());
    });

    it("on mintMessage should increase totalsupply and transfer amount", async () => {
        await stablecoinMasterContract.send(
            userPosition,
            { value: toNano(1) },
            { $$type: "MintMessage", user: user.address, amount: 100n }
        );

        await system.run();
        expect(track.collect()).toMatchSnapshot();

        expect(await stablecoinMasterContract.getTotalSupply()).toEqual(100n);
    });

    it("on repayBurn should send burn message to jetton", async () => {
        await stablecoinMasterContract.send(
            userPosition,
            { value: toNano(1) },
            { $$type: "RepayBurnMessage", user: user.address, amount: 100n, fees: 10n }
        );

        await system.run();
        expect(track.collect()).toMatchSnapshot();
    });

    it("on repayBurnNotification should send stablecoinBurned to user position", async () => {
        await stablecoinMasterContract.send(
            jetton,
            { value: toNano(1) },
            { $$type: "RepayBurnNotification", user: user.address, amount: 100n, fees: 10n }
        );

        await system.run();
        expect(track.collect()).toMatchSnapshot();
    });

    it("on MintFeesMessage should increase amount and send internaltransfer to jetton", async () => {
        await stablecoinMasterContract.send(
            gateKeeper,
            { value: toNano(1) },
            { $$type: "MintFeesMessage", to: user.address, amount: 100n }
        );

        await system.run();
        expect(track.collect()).toMatchSnapshot();

        expect(await stablecoinMasterContract.getTotalSupply()).toEqual(100n);
    });
});
