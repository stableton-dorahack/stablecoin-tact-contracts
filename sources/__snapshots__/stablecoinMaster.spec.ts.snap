// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StablecoinMasterContract on MintFeesMessage should increase amount and send internaltransfer to jetton 1`] = `
[
  {
    "$seq": 4,
    "events": [
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{9D42541D800784F79BB9BDDE39E2B05A2820A591FBCD92F3ACC083685CD65E37ED00489A4B22C9_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQBhDJZIy0_RS_IdyGG5GxdsC8eWb15DS9b3IL2p2-b0cUi8",
          "to": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
          "type": "internal",
          "value": 1000000000n,
        },
      },
      {
        "$type": "processed",
        "gasUsed": 16187n,
      },
      {
        "$type": "sent",
        "messages": [
          {
            "body": {
              "cell": "x{178D451900000000000000001648015E20D349B777BCD6E2B00BAFF49C4153BA9BE97DB5694F5B5448E768F8C7C189000F09EF37737BBC73C560B450414B23F79B25E7598106D0B9ACBC6FDA0091349642_}",
              "type": "cell",
            },
            "bounce": false,
            "from": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
            "to": "kQDIdzjMufp9FBvkEN02I0Kv1I9eMfkaQCDY7zHgfQ-Mij71",
            "type": "internal",
            "value": 968349000n,
          },
        ],
      },
    ],
  },
]
`;

exports[`StablecoinMasterContract on mintMessage should increase totalsupply and transfer amount 1`] = `
[
  {
    "$seq": 1,
    "events": [
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{903559B2800784F79BB9BDDE39E2B05A2820A591FBCD92F3ACC083685CD65E37ED00489A4B22C9_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQBqJ1A2wQ5MkOBlhdTr7ML850tmbmDecFDAEvNUwHvm0sGh",
          "to": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
          "type": "internal",
          "value": 1000000000n,
        },
      },
      {
        "$type": "processed",
        "gasUsed": 22427n,
      },
      {
        "$type": "sent",
        "messages": [
          {
            "body": {
              "cell": "x{178D451900000000000000001648015E20D349B777BCD6E2B00BAFF49C4153BA9BE97DB5694F5B5448E768F8C7C189000F09EF37737BBC73C560B450414B23F79B25E7598106D0B9ACBC6FDA0091349642_}",
              "type": "cell",
            },
            "bounce": false,
            "from": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
            "to": "kQDIdzjMufp9FBvkEN02I0Kv1I9eMfkaQCDY7zHgfQ-Mij71",
            "type": "internal",
            "value": 962109000n,
          },
        ],
      },
    ],
  },
]
`;

exports[`StablecoinMasterContract on repayBurn should send burn message to jetton 1`] = `
[
  {
    "$seq": 2,
    "events": [
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{B065C3AC800784F79BB9BDDE39E2B05A2820A591FBCD92F3ACC083685CD65E37ED00489A4B22C8215_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQBqJ1A2wQ5MkOBlhdTr7ML850tmbmDecFDAEvNUwHvm0sGh",
          "to": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
          "type": "internal",
          "value": 1000000000n,
        },
      },
      {
        "$type": "processed",
        "gasUsed": 19491n,
      },
      {
        "$type": "sent",
        "messages": [
          {
            "body": {
              "cell": "x{B065C3AC800784F79BB9BDDE39E2B05A2820A591FBCD92F3ACC083685CD65E37ED00489A4B22C8215_}",
              "type": "cell",
            },
            "bounce": false,
            "from": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
            "to": "kQDIdzjMufp9FBvkEN02I0Kv1I9eMfkaQCDY7zHgfQ-Mij71",
            "type": "internal",
            "value": 979086000n,
          },
        ],
      },
    ],
  },
]
`;

exports[`StablecoinMasterContract on repayBurnNotification should send stablecoinBurned to user position 1`] = `
[
  {
    "$seq": 3,
    "events": [
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{8C5F4C71800784F79BB9BDDE39E2B05A2820A591FBCD92F3ACC083685CD65E37ED00489A4B22C8215_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQCbJtY__64ruXlW7yhdMOa8y_YHO5HIIyKFXWvVWb58-ckR",
          "to": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
          "type": "internal",
          "value": 1000000000n,
        },
      },
      {
        "$type": "failed",
        "errorCode": 4429,
        "errorMessage": "Invalid sender",
      },
      {
        "$type": "sent-bounced",
        "message": {
          "body": {
            "type": "empty",
          },
          "bounce": false,
          "from": "kQCvEGmk27vea3FYBdf6TiCp3U30vtq0p62qJHO0fGPgxIkv",
          "to": "kQCbJtY__64ruXlW7yhdMOa8y_YHO5HIIyKFXWvVWb58-ckR",
          "type": "internal",
          "value": 990460000n,
        },
      },
    ],
  },
]
`;

exports[`StablecoinMasterContract should deploy correctly, set init state and premint amount to owner wallet 1`] = `
[
  {
    "$seq": 0,
    "events": [
      {
        "$type": "deploy",
      },
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{01FB345B000000000000000000000000000000000000000000000000000000746A5288004_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQAI-3FJVc_ywSuY4vq0bYrzR7S4Och4y7bTU_i5yLOB3A6P",
          "to": "kQAasLO_2xc6_cfOoccmU5-s13JSph3Ga85VBy0mBzgW4qbV",
          "type": "internal",
          "value": 1000000000n,
        },
      },
      {
        "$type": "processed",
        "gasUsed": 23821n,
      },
      {
        "$type": "sent",
        "messages": [
          {
            "body": {
              "cell": "x{178D451900000000000000005E8D4A510008003561677FB62E75FB8F9D438E4CA73F59AEE4A54C3B8CD79CAA0E5A4C0E702DC500023EDC525573FCB04AE638BEAD1B62BCD1ED2E0E721E32EDB4D4FE2E722CE07702_}",
              "type": "cell",
            },
            "bounce": false,
            "from": "kQAasLO_2xc6_cfOoccmU5-s13JSph3Ga85VBy0mBzgW4qbV",
            "to": "kQAcNvElz2dMtEP9esrAGOslGY8CrWfQLU1PaaZPmvHTPkvL",
            "type": "internal",
            "value": 951950000n,
          },
        ],
      },
    ],
  },
]
`;
